Feat-jobSearchFilter: Able to filter jobs on exisiting tags

Js-o opened the pull request:
For daily cron job, will only show jobs that match with the user's tags
Next steps would be for bot "job" trigger word to match jobs based on tags, and to not show repeats

Js-o committed:
Removed primary keys for users, user_jobs, user_Tags

only primary key is now the user id

Js-o committed:
Testing looping through arr for getting jobs

Need to make into promises

Js-o committed:
Updated Daily cron jobs to be filtered by your preferences

Will tell user to add tags if they do not have any

Js-o committed:
Changed to last jobs within 24 hours for daily cron jobs

Also made it so that the default jobs if user types in jobs (if not tags exist for user) will be javascript

Js-o committed:
Deleted console logs

Changed some lets to const

Js-o committed:
Removed extra console.log

Added note for tag[5] to indicate it is javascript

Js-o committed:
Changed back tags[5] in findJobsTag

back to tags

Js-o committed:
Simplified user_tag model and removed references inside define

Will simply have association by belongsToMany

pranaymohan commented on server/src/controllers/buttonController.js:
```diff
@@ -74,7 +74,7 @@ const saveJob = (req, res, data) => {
   })
   .catch(err => {
-    console.log('Error saving data');
+    console.log('Error saving data, ', err);
```
you dont need space after "data, " if we are using , in console log

pranaymohan commented on server/src/controllers/userTagController.js:
```diff
@@ -24,12 +28,12 @@ const addUserTags = (req, res) => {
   })
   .then(userTag => {
-    console.log('Destroyed, ', userTag.dataValues.userId, userTag.dataValues.tagId)
+    console.log('Destroyed, ', userTag.dataValues.slackUserId, userTag.dataValues.tagId)
```
dont need space after ,

pranaymohan commented on server/src/models/userModel.js:
```diff
@@ -18,7 +18,8 @@ let User = db.define('user', {
     type: Sequelize.STRING,
-    unique: true
+    primaryKey: true
+    // unique: true
```
remove this line

pranaymohan commented on server/src/utils/jobReminderCron.js:
```diff
@@ -5,9 +5,12 @@ import userJobsListener from '../bots/job';
 
 let jobCron = new CronJob({
   cronTime: '00 30 08 * * 1-5',
+  // cronTime: '50 * * * * *',
```
remove

pranaymohan commented on server/src/utils/jobReminderCron.js:
```diff
@@ -17,33 +20,58 @@ let jobCron = new CronJob({
-  User.findAll()
-  .then(users =>{
+  User.findAll({
+    include: [{
```
something feels weird abt this

pranaymohan commented on server/src/utils/jobReminderCron.js:
```diff
@@ -17,33 +20,58 @@ let jobCron = new CronJob({
+  })
+  .then(users => {
     users.forEach(user => {
+      let count ={};
```
space after =

pranaymohan commented on server/src/utils/jobReminderCron.js:
```diff
@@ -17,33 +20,58 @@ let jobCron = new CronJob({
     users.forEach(user => {
+      let count ={};
+      const tagArr = user.dataValues.tags.map(item => {
+        return item.name
```
;

pranaymohan commented on server/src/utils/jobReminderCron.js:
```diff
@@ -17,33 +20,58 @@ let jobCron = new CronJob({
-        include: [{ model: Job }],
+      user.dataValues.tags.forEach(tag => {
+        tag.jobs.forEach(job => {
+          if (count[job.id]) { count[job.id]++; }
```
please fix this structure -- one liner if statements with else blocks hurt my brain

pranaymohan commented on server/src/utils/jobReminderCron.js:
```diff
@@ -17,33 +20,58 @@ let jobCron = new CronJob({
+        BUCKLEY.startPrivateConversation({ user: user.slackUserId }, (err, convo) =>{
+          convo.say(`It seems like you don't have any tags! Please type tags and set you filters!`);
+          return;
+        })
```
;

pranaymohan commented on server/src/utils/jobReminderCron.js:
```diff
@@ -17,33 +20,58 @@ let jobCron = new CronJob({
+        BUCKLEY.startPrivateConversation({ user: user.slackUserId }, (err, convo) => {
+          convo.say(message_with_jobs);
+        });
+      })
```
no .catch block, but maybe best to refactor this big ol nested promise chain

Js-o committed:
Added recommendations for jobSearchFilter

Allowed cron time comment (for testing) and include to remain the same
Should refactor the nested promise when possible
